gym-unity/gym_unity/envs/__init__.py

```diff
@@ -151,12 +160,12 @@ def reset(self):
         self.game_over = False
 
         if not self._multiagent:
-            obs, reward, done, info = self._single_step(info)
+            res: GymStepResult = self._single_step(info)
         else:
-            obs, reward, done, info = self._multi_step(info)
-        return obs
+            res = self._multi_step(info)
```

Comment:
Is the type annotation only required for the 1st clause in the if statement?

Reply:
yeah, it's kind of annoying because mypy complains if you put it on both (even if it's the same type). 

The idea here is that _single_step and _multi_step return different types, but GymStepResult is the union of those two.